Sources/WalletSdk/ui/MRZScanner.swift

```diff
@@ -0,0 +1,457 @@
+
+            let requestHandler = VNImageRequestHandler(cvPixelBuffer: pixelBuffer, orientation: CGImagePropertyOrientation.up, options: [:])
+            do {
+                try requestHandler.perform([request])
+            } catch {
+                print(error)
+            }
+        }
+    }
+
+    func recognizeTextHandler(request: VNRequest, error: Error?) {
+
+        var codes = [String]()
+
+        guard let results = request.results as? [VNRecognizedTextObservation] else {
+            return
+        }
+
+        let maximumCandidates = 1
+        for visionResult in results {
+            guard let candidate = visionResult.topCandidates(maximumCandidates).first else { continue }
+
+            var numberIsSubstring = true
+
+            if let result = mrzFinder.checkMrz(str: candidate.string) {
+                if(result != "nil"){
+                    codes.append(result)
+                    numberIsSubstring = false
+                }
+            }
+        }
+
+        mrzFinder.storeAndProcessFrameContent(strings: codes)
+
+        // Check if we have any temporally stable numbers.
+        if let sureNumber = mrzFinder.getStableString() {
+            mrzFinder.reset(string: sureNumber)
+            scannedCode = sureNumber
+        }
+    }
+}
+
+
+class MRZFinder {
+    var frameIndex = 0
+    var captureFirst = ""
+    var captureSecond = ""
+    var captureThird = ""
+    var mrz = ""
+    var temp_mrz = ""
+
+    typealias StringObservation = (lastSeen: Int, count: Int)
+
+    // Dictionary of seen strings. Used to get stable recognition before
+    // displaying anything.
+    var seenStrings = [String: StringObservation]()
+    var bestCount = 0
+    var bestString = ""
+
+    func storeAndProcessFrameContent(strings: [String]) {
+        // Store all found strings
+        for string in strings {
+            if seenStrings[string] == nil {
+                seenStrings[string] = (lastSeen: 0, count: -1)
+            }
+            seenStrings[string]?.lastSeen = frameIndex
+            seenStrings[string]?.count += 1
+        }
+
+        // Remove all strings that weren't seen in a while
+        var obsoleteStrings = [String]()
+        for (string, obs) in seenStrings {
+            // Remove obsolete text after 30 frames (~1s).
+            if obs.lastSeen < frameIndex - 30 {
+                obsoleteStrings.append(string)
+            }
+
+            // Find the string with the greatest count.
+            let count = obs.count
+            if !obsoleteStrings.contains(string) && count > bestCount {
+                bestCount = count
+                bestString = string
+            }
+        }
+        // Remove old strings.
+        for string in obsoleteStrings {
+            seenStrings.removeValue(forKey: string)
+        }
+
+        frameIndex += 1
+    }
+
+    func checkMrz(str: String) -> (String)? {
+        let firstLineRegex = "(IAUT)(0|O)\\d{10}(SRC)\\d{10}<<"
```

Comment:
@cobward where can we find a spec about MDLs MRZ? Doesn't seems like it follows ICAO-9303.